### PR TITLE
commit

### PR DESCRIPTION
**Pull request by bohdanmayor**
[DEMO LINK](https://bohdanmayor.github.io/layout_html-form/)
[TEST REPORT LINK](https://bohdanmayor.github.io/layout_html-form/report/html_report/)

### CODE REVIEW

File: src/style.css [hunk 1]
```diff
@@ -1 +1,16 @@
...
+  margin: 20px 0;
+}
+
+form fieldset legend {
+  font-weight: bold;
+}
+
+form fieldset div {
+  padding: 5px 0;
+}
```
Comment (Viktor-Kost) [inline src/style.css]: use class selectors instead of tag ones

File: src/index.html [hunk 2]
```diff
@@ -11,7 +11,114 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="https://mate-academy-form-lesson.herokuapp.com/create-application" method="post" >
+      <fieldset>
+        <legend>Personal Information</legend>
+        <div class="personalInformation">
```
Comment (Viktor-Kost) [inline src/index.html]: no needs to use this wrapper `<div class="personalInformation">`

File: src/index.html [hunk 3]
```diff
@@ -11,7 +11,114 @@
...
+          <div class="personalInformationField">
+            <label for="surname">
+              Surname:
+            </label>
+            <input type="text" name="surname" id="surname">
+          </div>
```
Comment (Viktor-Kost) [inline src/index.html]: you can remove `<div class="personalInformationField">` wrapper. It's redundant. And refactor each of your input/label connection like this:

```suggestion
            <label class="field">
              Surname:
              <input 
                type="text" 
                name="surname" 
              >
            </label>
```

Apply all over the page

File: src/index.html [hunk 4]
```diff
@@ -11,7 +11,114 @@
...
+          <div class="personalInformationField">
+            <label for="name">
+              Name:
+            </label>
+            <input type="text" name="name" id="name">
```
Comment (Viktor-Kost) [inline src/index.html]: Text fields should have autocomplete="off"

File: src/index.html [hunk 5]
```diff
@@ -11,7 +11,114 @@
...
+          </div>
+          <div class="personalInformationField">
+            <label for="old">
+              How old are You?
+            </label>
+            <input type="number" id="old" name="old" value="12" min="1" max="100">
```
Comment (Viktor-Kost) [inline src/index.html]: if it has several attributes, you need to put each of them on its own line.

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Hello!)

Please add a DEMO link to a Pull Request description

**Comment by bohdanmayor** [comment]
hello, idk, but demo link and test report link dont working, so I didn't got appruve from first project with auto tests (hello-world...)

File: src/style.css [hunk 6]
```diff
@@ -1 +1,16 @@
 /* styles go here */
+body {
+  font-family: Arial, Helvetica, sans-serif;
+}
```
Comment (etojeDenys) [inline src/style.css]: move to html selector

**Review by etojeDenys — CHANGES_REQUESTED** [review]
you can ask about how to deploy the project and add links to PR in the slack chat

**Review by etojeDenys — CHANGES_REQUESTED** [review]
add links to your PR or feel free to ask about that in the slack chat

**Comment by bohdanmayor** [comment]
> add links to your PR or feel free to ask about that in the slack chat

what it is PR?

File: readme.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # HTML form
 Replace `<your_account>` with your Github username and copy the links to Pull Request description:
-- [DEMO LINK](https://<your_account>.github.io/layout_html-form/)
-- [TEST REPORT LINK](https://<your_account>.github.io/layout_html-form/report/html_report/)
+- [DEMO LINK](https://bohdanmayor.github.io/layout_html-form/)
+- [TEST REPORT LINK](https://bohdanmayor.github.io/layout_html-form/report/html_report/)
```
Comment (DarkMistyRoom) [inline readme.md:4]: add TEST REPORT LINK to the description too

File: src/style.css [hunk 8]
```diff
@@ -1 +1,16 @@
...
+
+.registerForm fieldset {
+  margin: 20px 0;
+}
+
+.registerForm fieldset legend {
+  font-weight: bold;
+}
```
Comment (DarkMistyRoom) [inline src/style.css]: Remember, any other styles besides margins, should be browser default.

Comment (DarkMistyRoom) [reply src/style.css]: image

File: src/index.html [hunk 9]
```diff
@@ -11,7 +11,154 @@
...
+          id="no"
+          name="love-cats"
+          value="No"
+          >
+          <label for="no">No</label>
+        </div>
+        <div>
+          What is your favourite color?
+          <input type="color">
+        </div>
```
Comment (DarkMistyRoom) [inline src/index.html]: use label for this element too

Comment (DarkMistyRoom) [reply src/index.html]: you should use it for all fields. check it everwhere

File: src/index.html [hunk 10]
```diff
@@ -11,7 +11,154 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form class="registerForm"
+    action="https://mate-academy-form-lesson.herokuapp.com/create-application"
+    method="post"
+    >
```
Comment (DarkMistyRoom) [inline src/index.html]:
```suggestion
    <form 
      class="registerForm"
      action="https://mate-academy-form-lesson.herokuapp.com/create-application"
      method="post"
    >
```

Comment (DarkMistyRoom) [reply src/index.html]: keep the code style

Comment (DarkMistyRoom) [reply src/index.html]: fix it everywhere

File: src/index.html [hunk 11]
```diff
@@ -11,7 +11,154 @@
...
+            Surname:
+            <input
+            type="text"
+            autocomplete="off"
+            name="surname"
+            id="surname"
+            >
+          </label>
+        </div>
+        <div>
```
Comment (DarkMistyRoom) [inline src/index.html]: Add empty lines between multiline sibling blocks of HTML

File: src/style.css [hunk 12]
```diff
@@ -1 +1,16 @@
 /* styles go here */
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.registerForm fieldset {
+  margin: 20px 0;
```
Comment (DarkMistyRoom) [inline src/style.css]: use one-direction margin. only margin-top or margin-bottom

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
there should be required fields.

Please write in the chat if you have questions

File: src/index.html [hunk 13]
```diff
@@ -11,7 +11,178 @@
...
+        <legend>An interesting fact about you!</legend>
+        <div class="field">
+          <label for="yes">
+            Do you love cats?
+            <input
+            type="radio"
+            id="yes"
+            name="love-cats"
+            value="yes"
+            >
```
Comment (IvanFesenko) [inline src/index.html]: Pay attention on indents
```suggestion
            <input
              type="radio"
              id="yes"
              name="love-cats"
              value="yes"
            >
```

File: src/style.css [hunk 14]
```diff
@@ -1 +1,11 @@
-/* styles go here */
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.registerForm .fieldset {
+  margin-top: 20px;
+}
```
Comment (IvanFesenko) [inline src/style.css]: You can just use `.fieldset`
```suggestion
.fieldset {
  margin-top: 20px;
}
```

File: src/index.html [hunk 15]
```diff
@@ -11,7 +11,178 @@
...
+            How do you rate our work?
+            <input type="range" id="workRate">
+          </label>
+        </div>
+      </fieldset>
+      <fieldset class="fieldset">
+        <legend>Additonal info:</legend>
+        <div class="field">
+
+        </div>
```
Comment (IvanFesenko) [inline src/index.html]: Remove empty div

File: src/style.css [hunk 16]
```diff
@@ -1 +1,11 @@
...
+
+.registerForm .fieldset .field {
+  padding-top: 5px;
+}
```
Comment (IvanFesenko) [inline src/style.css]: Same here

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
It should be multi select. Also check my comments

File: src/style.css [hunk 17]
```diff
@@ -1 +1,11 @@
-/* styles go here */
+html {
+  font-family: Arial, Helvetica, sans-serif;
```
Comment (mykhalenych) [inline src/style.css]: I think it is redundant, and don't see in requirements anything about font-family

File: src/index.html [hunk 18]
```diff
@@ -11,7 +11,176 @@
...
+        <div class="field">
+          <label for="carBrand">
+            What are your favourite brend of cars?
+            <select
+              name="carBrand"
+              id="carBrand"
+              required
+              multiple
+            >
+              <option disabled selected>Chose</option>
```
Comment (mykhalenych) [inline src/index.html]: delete it

File: src/index.html [hunk 19]
```diff
@@ -11,7 +11,176 @@
...
+              <option value="chevrolet">Chevrolet</option>
```
Comment (mykhalenych) [inline src/index.html]: I see that on design
Screenshot 2023-03-17 at 16 59 54

**Review by mykhalenych — CHANGES_REQUESTED** [review]
check comments, also check tests you failed 7 tests

File: src/style.css [hunk 20]
```diff
@@ -1 +1,17 @@
...
+  padding-bottom: 10px;
+}
+
+#date {
+  width: 148px;
+}
+
+#time {
+  width: 98px;
+}
```
Comment (anstsot) [inline src/style.css]: these styles are redundant
```suggestion
```

**Review by anstsot — CHANGES_REQUESTED** [review]
The code looks good but the tests are failed. could you please check them?
If you have any questions or troubles, feel free to ask them in the chat

File: src/style.css (hunk 20 above)
Comment (bohdanmayor) [reply src/style.css]: изображение
all tests passed

Comment (bohdanmayor) [reply src/style.css]: I understood, I have to change gh-pages (I have all tests passed localy, and terminal, but that file is old)

Comment (bohdanmayor) [reply src/style.css]: Idk how to deploy, I have deploy error

Comment (bohdanmayor) [reply src/style.css]: can you appruve it without gh-pages?

**Review by mykhalenych — CHANGES_REQUESTED** [review]
your links aren't working

**Comment by bohdanmayor** [comment]
I know, i cant deploy pages, i have an error on npm deploy

**Pull request closed without merging**